tools/block-generator/README.md

```diff
@@ -204,5 +207,133 @@ block-generator runner \
+
+## Scenario Distribution - Configuration vs. Reality
+
+This section follows up on the [Scenario Configuration](#scenario-configuration) section to detail how each kind of transaction is actually chosen.
+Note that -especially for early rounds- there is no guarantee that the
+percentages of transaction types will resemble the configured distribution.
+
+For example consider the [Organic 25,000](scenarios/benchmarks/organic.25000.yml) scenario:
+
+```yml
+name: "Organic (25000)"
+genesis_accounts: 10000
+genesis_account_balance: 1000000000000
+tx_per_block: 25000
+
+# transaction distribution
+tx_pay_fraction: 0.05
+tx_asset_fraction: 0.75
+tx_app_fraction: 0.20
+
+# payment config
+pay_acct_create_fraction: 0.10
+pay_xfer_fraction: 0.90
+
+# asset config
+asset_create_fraction: 0.001
+asset_optin_fraction: 0.1
+asset_close_fraction: 0.05
+asset_xfer_fraction: 0.849
+asset_delete_fraction: 0
+
+# app kind config
+app_boxes_fraction: 1.0
+app_swap_fraction: 0.0
+
+# app boxes config
+app_boxes_create_fraction: 0.01
+app_boxes_optin_fraction: 0.1
+app_boxes_call_fraction: 0.89
+```
+
+We are _actually_ asking the generator for the following distribution:
+
+* `pay_acct_create_fraction = 0.005 (= 0.05 * 0.10)`
+* `pay_xfer_fraction =  0.045 (= 0.05 * 0.90)`
+* `asset_create_fraction = 0.00075 (= 0.75 * 0.001)`
+* `asset_optin_fraction = 0.075 (= 0.75 * 0.1)`
+* `asset_close_fraction = 0.0375 (= 0.75 * 0.05)`
+* `asset_xfer_fraction = 0.63675 (= 0.75 * 0.849)`
+* `asset_delete_fraction = 0`
+* `app_boxes_create_fraction = 0.002 (= 0.20 * 1.0 * 0.01)`
+* `app_boxes_optin_fraction = 0.02 (= 0.20 * 1.0 * 0.1)`
+* `app_boxes_call_fraction = 0.178 (= 0.20 * 1.0 * 0.89)`
+
+The block generator randomly chooses
+
+1. the transaction type (pay, asset, or app) according to the `transaction distribution`
+2. based on the type:
+
+   a. for payments and assets, the specific type based on the `payment config` and `asset config` distributions
+
+   b. for apps, the app kind (boxes or swaps) based on the `app kind config` distribution
+
+3. For _apps only_: the specific app call based on the `app boxes config` (and perhaps in the future `app swap config`)
+
+As each of the steps above is itself random, we only expect _approximate matching_ to the configured distribution.
+
+Furthermore, for certain asset and app transactions there may be a substitution that occurs based on the type. In particular:
+
+* for **assets**:
+  * when a requested asset txn is **create**, it is never substituted
+  * when there are no assets, an **asset create** is always substituted
+  * when a requested asset txn is **delete** but the creator doesn't hold all asset funds, an **asset close** is substitued (which itself may be substituted using the **close** rule below)
+  * when a requested asset txn is **opt in** but all accounts are already opted in, an  **asset close** is substituted (which itself may be substituted using the **close** rule below)
```

Comment:
Is this true? I would have expected a new account to be created or a new asset to be selected.

Reply:
I just followed the present logic, but I'll double check.

Reply:
> when a requested asset txn is **create**, it is never substituted
[line 660](https://github.com/algorand/go-algorand/blob/8281f8abcbd2ecdc8937e2d61320f84a57ca9ada/tools/block-generator/generator/generate.go#L660)

> when there are no assets, an **asset create** is always substituted
[line 649](https://github.com/algorand/go-algorand/blob/8281f8abcbd2ecdc8937e2d61320f84a57ca9ada/tools/block-generator/generator/generate.go#L649)

> when a requested asset txn is **delete** but the creator doesn't hold all asset funds ...
[line 692](https://github.com/algorand/go-algorand/blob/8281f8abcbd2ecdc8937e2d61320f84a57ca9ada/tools/block-generator/generator/generate.go#L692)

> when a requested asset txn is **opt in** but all accounts are already opted in ...
[line 708](https://github.com/algorand/go-algorand/blob/8281f8abcbd2ecdc8937e2d61320f84a57ca9ada/tools/block-generator/generator/generate.go#L708)